tests/test-iso-kargs.sh

```diff
@@ -0,0 +1,62 @@
+#!/bin/bash
+set -xeuo pipefail
+
+fatal() {
+    echo "$@" >&2
+    exit 1
+}
+
+iso=$1; shift
+iso=$(realpath "${iso}")
+
+tmpd=$(mktemp -d)
+trap 'rm -rf "${tmpd}"' EXIT
+cd "${tmpd}"
+
+cp --reflink=auto "${iso}" "test.iso"
+iso=test.iso
+orig_hash=$(sha256sum "${iso}")
+
+# Sanity-check the ISO doesn't somehow already have the karg we're testing with.
+if coreos-installer iso kargs show "${iso}" | grep -q foobar; then
+    fatal "Unexpected foobar karg in iso kargs"
+fi
+
+# Test all the modification operations.
+coreos-installer iso kargs modify -a foobar=oldval "${iso}"
+coreos-installer iso kargs show "${iso}" | grep -q foobar=oldval
+coreos-installer iso kargs modify -a dodo "${iso}"
+coreos-installer iso kargs show "${iso}" | grep -q 'foobar=oldval dodo'
+coreos-installer iso kargs modify -r foobar=oldval=newval "${iso}"
+coreos-installer iso kargs show "${iso}" | grep -q 'foobar=newval dodo'
+coreos-installer iso kargs modify -d foobar=newval "${iso}"
+if coreos-installer iso kargs show "${iso}" | grep -q 'foobar'; then
+    fatal "Unexpected foobar karg in iso kargs"
+fi
+coreos-installer iso kargs show "${iso}" | grep -q 'dodo'
+coreos-installer iso kargs modify -d dodo "${iso}"
+
+hash=$(sha256sum "${iso}")
+if [ "${orig_hash}" != "${hash}" ]; then
+    fatal "Hash doesn't match original hash: ${hash} vs ${orig_hash}"
+fi
+
+# Test the largest karg; this technically is a detail of the padding in the
+# distro configs. Here we're assuming an FCOS/RHCOS ISO: 1024 - 2 (2 for the
+# final '\n#' to put "COREOS_KARG_EMBED_AREA on the next line).
```

Comment:
:gift:

```
readu8() {
    dd if="${iso}" bs=1 skip="$1" count=8 status=none | od --endian=little -t d8 | head -n1 | awk '{print $2}'
}
embed_size=$(readu8 32680)
embed_used_size=$(dd if="${iso}" bs=1 skip=$(readu8 32688) count=${embed_size} status=none | grep -bo "#" | head -n 1 | cut -f1 -d:)
embed_usable_size=$((${embed_size} - ${embed_used_size} - 1))
```

Reply:
Heh thanks! I liked the idea of just getting it from the ISO instead of assuming a value. I ended up taking it a bit further and adding a `--header` hidden switch so we just get it from `coreos-installer` itself.